### ISSUE TITLE
Error with versions of Node.js below v10

### ISSUE THREAD

**Issue by M-casado** [elixir-europe/biovalidator#27]
### Summary
Executing tests (``npm test``) of Biovalidator using nodejs ``v8.11.1`` throws syntax errors at ``biovalidator/node_modules/jest/node_modules/jest-cli/build/cli/index.js:227``

## Description
Hi, I am trying to make use of the Biovalidator in the CLI using ``validator-cli.js`` to validate a JSON document against a custom schema (Draft 2019-09). 
After installing Biovalidator using nodejs ``v8.11.1`` (npm ``v5.6.0``) and performing the tests it throws the following error:
````
$ npm test

> biovalidator@1.9.1 test /mnt/c/Users/mcasado/Documents/GitHub/biovalidator
> jest

/mnt/c/Users/mcasado/Documents/GitHub/biovalidator/node_modules/jest/node_modules/jest-cli/build/cli/index.js:227
    } catch {
            ^

SyntaxError: Unexpected token {
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:616:28)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/mnt/c/Users/mcasado/Documents/GitHub/biovalidator/node_modules/jest/node_modules/jest-cli/bin/jest.js:16:3)
npm ERR! Test failed.  See above for more details.
````

Making use of the ``validator-cli.js`` with this erroneous build unveils inconsistencies between ``nodejs`` (``v8.11.1``) and ``yargs``.
````
$ time node ./validator-cli.js -s examples/schemas/test-schema.json -j examples/objects/test-schema-valid.json
/mnt/c/Users/mcasado/Documents/GitHub/biovalidator/node_modules/yargs-parser/build/index.cjs:997
        throw Error(`yargs parser supports a minimum Node.js version of ${minNodeVersion}. Read our version support policy: https://github.com/yargs/yargs-parser#supported-nodejs-versions`);
        ^

Error: yargs parser supports a minimum Node.js version of 10. Read our version support policy: https://github.com/yargs/yargs-parser#supported-nodejs-versions
    at Object.<anonymous> (/mnt/c/Users/mcasado/Documents/GitHub/biovalidator/node_modules/yargs-parser/build/index.cjs:997:15)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/mnt/c/Users/mcasado/Documents/GitHub/biovalidator/node_modules/yargs/build/index.cjs:2855:16)
    at Module._compile (module.js:652:30)

real    0m0.094s
user    0m0.053s
sys     0m0.000s
````

### AJV execution
When trying to validate the JSON document against its schema using AJV directly (its CLI feature ``ajv-cli``) it does work and validation is passed:
````
$ ajv --spec=draft2019 -s schema.json -d json_doc.json
strict mode: missing type "object" for keyword "additionalProperties" at "test_schema_X#/properties/sample_labels" (strictTypes)
json_doc.json valid
````
### Using nodejs v10.19.0
Tests are passed correctly:
````
Test Suites: 10 passed, 10 total
Tests:       27 passed, 27 total
Snapshots:   0 total
Time:        48.545 s
Ran all test suites.
````

And using ``validator-cli.js`` with test schemas and documents seems to work:
````
$ time node ./validator-cli.js -s examples/schemas/test-schema.json -j examples/objects/test-schema-valid.json
 No validation errors reported.
Validation finished.

real    0m6.116s
user    0m0.867s
sys     0m0.571s

$ time node ./validator-cli.js -s examples/schemas/test-schema.json -j examples/objects/test-schema-invalid.json
 The validation process has found the following error(s):

 /characteristics.organism
        should have required property 'organism'
/characteristics.Organism
        should have required property 'Organism'
/characteristics.species
        should have required property 'species'
/characteristics.Species
        should have required property 'Species'
/characteristics
        should match some schema in anyOf

Validation finished.

real    0m6.076s
user    0m0.813s
sys     0m0.595s
````

### SO and versions
The details of the build that passes the tests for me are:
* **Windows Subsystem for Linux 2** (WSL2)
* ajv-cli ``v5.0.0``
* npm ``v6.13.4`` 
* And nodejs details are:
````
$ node -p process.versions
{ http_parser: '2.9.3',
  node: '10.19.0',
  v8: '6.8.275.32-node.55',
  uv: '1.28.0',
  zlib: '1.2.11',
  brotli: '1.0.7',
  ares: '1.15.0',
  modules: '64',
  nghttp2: '1.39.2',
  napi: '5',
  openssl: '1.1.1d',
  icu: '64.2',
  unicode: '12.1',
  cldr: '35.1',
  tz: '2019c' }
````

I also tried using the latest versions of npm (``v8.1.3``) and nodejs (``v16.13.0``) which had a similar behaviour as the versions I managed to make biovalidator work with.

The only issue in working with versions ``>10`` (that seem to work) of nodejs is that there is an additional issue (related to memory use) I found when validating my custom document and schema. But this can be traced in another GH issue if necessary.

**Comment by theisuru** [comment]
thanks @M-casado for pointing this. It seems like in recent changes we have done, we used features of newer node versions (such as BigInt introduced in 10.4+). Hence this fails in the version we have listed in the README file. I will change the supported version in the README to v10.19.0 which is working for me.
We appriciate if you can open another issue for memory use related problem you have mentioned.

**Comment by M-casado** [comment]
@theisuru I would like to point out that the change was merged into ``dev`` but not ``main``. Should I be using ``dev`` instead? I suppose everyone would be using the ``main`` stable version, so that's what I'm currently using. But the versions are giving me issues again, seeing the following message when I execute ``npm install``:
````
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'biovalidator@1.9.1',
npm WARN EBADENGINE   required: { node: '8.11.1' },
npm WARN EBADENGINE   current: { node: 'v16.13.0', npm: '8.3.0' }
npm WARN EBADENGINE }
````
Is Biovalidator not supposed to work with newer versions of ``node``, following the nomenclature of your changes ``v10.19.1.0+``? It is currently quite troublesome to manage versions of ``node`` and ``npm`` following the main ``README``. 

Besides, is there a way of retrieving the versions of ``node`` and ``npm`` that were used when installing Biovalidator? For instance, I currently still have Biovalidator installed locally at some place, but I can't remember what versions I used when installing it, and since I started using ``AJV-cli`` I updated ``node`` and ``npm`` to their latest versions. 

In any case, if possible it would be awesome for the tool to work with the newest versions of those two packages, since trying to ``npm install`` it with them throws some warnings and finally errors.
